src/exoticatechnologies/hullmods/exotics/HullmodExoticHandler.kt

```diff
@@ -18,183 +29,856 @@ object HullmodExoticHandler {
+//                        mutableVariantList.add(member.variant)  //TODO get rid of this, we should probably just rename this into childModulesVariantList
+                        mutableVariantList.add(memberVariant)
+
+                        mutableVariantList.toList()
+                    }
+
+                    // Return the optional of the list
+                    Optional.of(variants)
+                }
+
+                val shouldInstallOnMemberVariant = HullmodExoticHandler.shouldInstallHullmodExoticToVariant(
+                        hullmodExotic = hullmodExotic,
+                        parentFleetMember = member,
+                        variant = memberVariant,
+                        variantList = variantListOptional,
+                        workMode = workMode
+                )
+                onShouldCallback.execute(shouldInstallOnMemberVariant, memberVariant)
+                logger.info("onInstall()\tshouldInstallOnMemberVariant: ${shouldInstallOnMemberVariant}, variant: ${memberVariant}")
+                if (shouldInstallOnMemberVariant) {
+                    val installResult = HullmodExoticHandler.installHullmodExoticToVariant(
+                            hullmodExotic = hullmodExotic,
+                            parentFleetMember = member,
+                            variant = memberVariant,
+                            workMode = workMode
+                    )
+                    onInstallCallback.execute(installResult, memberVariant)
+                }
+            }
+
+            @JvmStatic
+            fun CheckAndRemoveFromAllChildModulesVariants(
+                    fleetMember: FleetMemberAPI,
+                    hullmodExotic: HullmodExotic,
+                    onShouldCallback: OnShouldCallback,
+                    onRemoveFromChildModuleCallback: OnRemoveFromChildModuleCallback
+            ) {
+                // First things first, figure out whether we're running from Refit or Exotica screen
+                val isFromRefitScreen = runningFromRefitScreen()
+                val workMode = if (isFromRefitScreen) {
+                    HullmodExoticHandlerWorkMode.LENIENT
+                } else {
+                    HullmodExoticHandlerWorkMode.STRICT
+                }
+
+                //Lets go through all 'installed modules' and uninstall from them, instead of these module slots
+                // that were the root cause of the whole problem... damn you 3-5am copypasted code ...
+                val hullmodExoticKey = HullmodExoticKey(
+                        hullmodExotic = hullmodExotic,
+                        parentFleetMemberId = fleetMember.id
+                )
+                val currentInstallData = synchronized(lookupMap) { lookupMap[hullmodExoticKey] }
+                val installedOnVariantsList = currentInstallData?.listOfVariantsWeInstalledOn
+                        ?: throw IllegalStateException("No InstallData found for key ${hullmodExoticKey}")
```

Comment:
while i felt smart doing it like this, i'd much rather have a 
```
if (currentInstallData != null) {
    curentInstallData.listOfVariantsWeInstalledOn
} else {
    throw ISE("blabla")
}
```

Reply:
oh the `throw` part should go, since trying to remove it from e.g. child module will crash the game with "no install data found" instead of just not doing anything.

Reply:
done

Reply:
resolving.